src/Symfony/Component/Form/Tests/Extension/Core/EventListener/ResizeFormListenerTest.php

```diff
@@ -255,7 +255,7 @@ public function testOnSubmitDeleteEmptyNotCompoundEntriesIfAllowDelete()
 
         $data = [0 => 'first', 1 => ''];
         foreach ($data as $child => $dat) {
-            $this->form->get($child)->setData($dat);
+            $this->form->get($child)->submit($dat);
```

Comment:
this change (and the next below) is related to `$child->isValid()` check inside `onSubmit()` method. It's expected anyway.

Reply:
Wouldn't it make more sense if we called `submit()` on `$this->form` instead?

Reply:
Normally yes, but we can't in this test case because this exception "You cannot remove children from a submitted form." when calling to `$listener->onSubmit($event)` (some lines below).

Reply:
We could remove the `onSubmit()` call as it will be called implicitly when `submit()` is called, isn't it?

Reply:
It won't be called, it's not the CollectionType form test.

Reply:
This test case is simulating the usage of the ResizeFormListener submittion on any compound form.